# Primer día aprendiendo a manejar Python

## Estoy en clase de Python (en Data Science) y estoy probando diferentes cositas

In [1]:
2+2

4

# Prueba 1

## Subtexto

### subsubtexto

# Declarando multiples variables en una sola línea

In [18]:
age, name, city, world = 29, "Irene", "Madrid", "la Tierra"

In [19]:
print(world)

la Tierra


In [20]:
print(name," ",city)

Irene   Madrid


In [21]:
tres=three=drei=3

In [22]:
print(three)

3


# Conocer tipo variable

In [35]:
type(3)

int

# Jugar con posiciones

In [7]:
print("hello\n\tWorld")

hello
	World


In [25]:
print("     *\n*\t*\n    *")

     *
*	*
    *


# Asignar variable a otra variable

In [28]:
Gente_en_clase = 16


In [32]:
Total_population = Gente_en_clase

In [34]:
print(Total_population)

16


 Para poder relacionar dos nombres como "Gente_en_clase" y "Total_Population" la segunda línea, debo poner primero "Total_population" porque tengo que definirla (la otra ya está definida y asignada a num 16), la voy a definir ahí es T_p así que es la que va primero. Si no, no funciona.
Y tiene que estar escrita tal cual,las mayus y minus influyen.

# Operando con variable numérica

In [37]:
x = 2314
x+1

2315

In [39]:
y=x+2314
y

4628

# Sobreescribir variable num (2 formas)

In [40]:
x = x * 0.1

In [41]:
x

231.4

Esto (abajo) se llama: *Asignación aumentada*.

In [42]:
x +=0.5
x

231.9

In [43]:
x -=31
x

200.9

 *Asignación tradicional* es cuando simplemente se vuelve a renombrar la variable ej. x = 5
x = x + 1

In [92]:
random_number = 15
random_number

15

In [93]:
random_number += 5
random_number

20

In [94]:
random_number *=5
random_number

100

# Strings

Variable se identifica con minusculas 'x'. Cuando es una constante, algo que no debo cambiar, se pone en MAYUSCULAS, un dato que no debo reasignar,resignificar. No hay otra forma.

In [13]:
a = 'Hola'
b = 'Mundo'
e= ' '

In [14]:
print(a+e+b)

Hola Mundo


In [15]:
print(a+" "+b)

Hola Mundo


## Concatenación

In [49]:
print("Mi nombre es "+ name + " y vivo en " + city)

Mi nombre es Irene y vivo en Madrid


# Interpolación
### más usado y cómodo

In [45]:
print("Mi nombre es {} y vivo en {}".format(name,city))

Mi nombre es Irene y vivo en Madrid


# Count

In [51]:
print(name.count("e"))

2


In [52]:
print(len(name))

5


In [55]:
print(city.upper())

MADRID


# Actividad 1

In [56]:
space_oddysey = 2001
space_oddysey

2001

In [57]:
type('Good night & Good luck')

str

In [58]:
type(True)

bool

In [61]:
result= 10*3.0

In [62]:
result

30.0

In [63]:
type(result)

float

# Seguimos!
## operaciones
Para sumar numeros enteros int y float, pero quiero el resultado en enteros, debo poner "int" antes de la operación entre paréntesis. Lo mismo al revés si quiero que aparezca en float..porque sí.

In [67]:
int (8 + 4.2)

12

In [68]:
8 + 4.2

12.2

In [69]:
float (8+7)

15.0

In [74]:
10/3

3.3333333333333335

In [73]:
10//3

3

Formas de elevar a la n

In [71]:
5**5

3125

In [72]:
pow (5,5)

3125

# Num complejos (complex)

In [75]:
z= 2 + 5j
z

(2+5j)

In [76]:
type(z)

complex

La i siempre vale raiz de -1, no se usa para ecuaciones/funciones. Significa que hay información que no procesamos, "ruido", info adicional. Si es 2 + 5j, significa que hay 2 unidades de x que sí podemos explicar/procesar y 5 unidades de "ruido", de datos extra que no podemos identificar.

In [77]:
pip install colorama

Note: you may need to restart the kernel to use updated packages.


Me da la parte real solo del numero completo // o solo la parte imaginaria

In [80]:
z.real

2.0

In [81]:
z.imag

5.0

In [82]:
o= z + (3 +5j)
o


(5+10j)

In [83]:
o*z

(-40+45j)

Para calcular el modulo del numero complejo, el valor absoluto. Lo convierte en positivo si está en negativo y elimina la "j"

In [84]:
abs (z)

5.385164807134504

Todas las operaciones que quiera hacer sobre los ocmplejos, se puede hacer con cmath (c de complejo)

In [88]:
import math as mt

In [86]:
import cmath

In [87]:
cmath.phase(z)

1.1902899496825317

Al hacer esta operación de fase, que dice el ángulo que hay entre la línea imaginaria y la línea real, el º del ángulo nos dice dónde está la informacion. Fase cercana a 90º significa que están casi toda la info en el eje imaginario y que estamos perdiendo casi toda la info, y si está más cerca de 0º, es que estamos más cerca del eje real y podemos saber la info.La info anterior 1.1903... está en radianes y hay que pasarlo a grados.

In [89]:
z

(2+5j)

In [90]:
cmath.polar (z)

(5.385164807134504, 1.1902899496825317)

El 5.38 es el módulo y el 1.19 es la fase.

## Valor absoluto. Pasa cualquier num negativo a abs

In [95]:
abs(-1.2)

1.2

# Sumar con Booleanos + Conversiones implícitas
Python da prioridad al numero entero o real etc y lo convierte implícitamente (conversion implícita) para evitar errores, en la misma categoría de datos

In [99]:
True + 25.01

26.01

In [97]:
True + True

2

In [98]:
7* False

0

# Conversiones explícitas. 
### Transformar bool,float, int en otro tipo. 

In [100]:
bool(0)

False

In [101]:
bool(-1)

True

In [102]:
bool(2)

True

In [103]:
bool(3)

True

In [104]:
int(True)

1

In [105]:
float (34)

34.0

In [106]:
int(3.4554)

3

In [109]:
int (8.9)

8

Cuidado con no abusar de esto,porque nos da la parte entera, pero no redondea. Es decir, estamos perdiendo un 0.9 de información que es mucho.

In [110]:
(19/115)*(115/19)

0.9999999999999999

También pueden dar errores de aproximación, al ser una division flotante, el resultado da decimal. Sin embargo, si lo hacemos por calculadora, esa multiplicación saldría "1". Salen errores.

# Round
### te redondea el número, utiliza la regla de redondea a mayor a partir del 5

In [111]:
from math import pi, log, exp

In [112]:
pi

3.141592653589793

In [113]:
round(pi)

3

In [114]:
round(pi,2)

3.14

In [115]:
round((19/115)*(115/19))

1

## Límite de un flotante. Sí hay límite. Y aparece así:

In [116]:
import sys

In [117]:
sys.float_info.min

2.2250738585072014e-308

In [118]:
sys.float_info.max

1.7976931348623157e+308